orchestration/include/forge/orch/ResearchNode.hpp

```
@@ -0,0 +1,140 @@
+public:
+  bool review(const retrieval::QueryPreview& preview) override { return preview.sendable(); }
+  std::string approverName() const override { return "headless-auto"; }
+};
+
+class ResearchNode {
+public:
+  ResearchNode(std::shared_ptr<retrieval::SearxngClient> client,
+               std::shared_ptr<ApprovalProvider> approver,
+               CheckpointStore* checkpoints);
+
+  static const char* kNodeId;
```

Clang (14.0.6)

[warning] variable 'kNodeId' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)